src/providers/NotificationProvider.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { createContext, useContext, useState, ReactNode } from 'react';
+import { Notification as StellarNotification } from '@stellar/design-system';
+
+type NotificationType = 'primary' | 'secondary' | 'success' | 'error' | 'warning';
+interface Notification {
+  message: string;
+  type: NotificationType;
+  isVisible: boolean;
+}
+
+interface NotificationContextType {
+  addNotification: (message: string, type: NotificationType) => void;
+}
+
+const NotificationContext = createContext<NotificationContextType | undefined>(undefined);
+
+export const NotificationProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
+  const [notifications, setNotifications] = useState<Notification[]>([]);
+
+  const addNotification = (message: string, type: NotificationType) => {
+    const newNotification = { message, type, isVisible: true };
+    setNotifications((prev) => [...prev, newNotification]);
+
+    setTimeout(() => {
+      setNotifications((prev) =>
+        prev.map((notification) =>
+          notification === newNotification ? { ...notification, isVisible: false } : notification
+        )
+      );
+    }, 2500); // Start fade-out after 2.5 seconds
+
+    setTimeout(() => {
+      setNotifications((prev) => prev.filter((notification) => 
+        !(notification.message === newNotification.message && 
+          notification.type === newNotification.type)
+      ));
+    }, 3000); // Remove after 3 seconds
+  };
+
+  return (
+    <NotificationContext.Provider value={{ addNotification }}>
+      {children}
+      <div style={{ position: 'fixed', top: '10px', left: '50%', transform: 'translateX(-50%)', zIndex: 1000 }}>
+        {notifications.map((notification, index) => (
+          <div
+            key={index}
+            style={{
+              opacity: notification.isVisible ? 1 : 0,
+              transition: 'opacity 0.5s ease',
```

Comment:
If I'm understanding correctly, if the app shows multiple notifications quickly, the top one will fade out in 0.5s while the one below it stays still, then suddenly after 2.5 more seconds, the bottom one will jump, not slide, to the top. Is that right?

Since both timings are set in the same file, I wonder if we should use a variable to define both.

We could also fade with `opacity` and then slide with `top`? Or do them at the same time? `transition` both of them.

Reply:
Probably can do transition things! Or transition with height to 0 maybe?